// ngx-croppie/esm2015/src/app/modules/ngx-croppie/ngx-croppie.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component, Input, EventEmitter, Output, ViewChild, ElementRef } from '@angular/core';
import * as Croppie from 'croppie';
export class NgxCroppieComponent {
    constructor() {
        this.outputFormatOptions = { type: 'base64', size: 'viewport' };
        this.defaultZoom = 0;
        this.result = new EventEmitter();
    }
    /**
     * @return {?}
     */
    get imageUrl() {
        return this.imgUrl;
    }
    /**
     * @param {?} url
     * @return {?}
     */
    set imageUrl(url) {
        if (this.imgUrl === url) {
            return;
        }
        this.imgUrl = url;
        if (this._croppie) {
            this.bindToCroppie(this.imageUrl, this.points, this.defaultZoom);
        }
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        this._croppie = new Croppie(this.imageEdit.nativeElement, this.croppieOptions);
        this.bindToCroppie(this.imageUrl, this.points, this.defaultZoom);
    }
    /**
     * @param {?} url
     * @param {?} points
     * @param {?} zoom
     * @return {?}
     */
    bindToCroppie(url, points, zoom) {
        this._croppie.bind({ url, points, zoom });
    }
    /**
     * @return {?}
     */
    newResult() {
        this._croppie.result(this.outputFormatOptions).then((res) => {
            this.result.emit(res);
        });
    }
    /**
     * @param {?} degrees
     * @return {?}
     */
    rotate(degrees) {
        this._croppie.rotate(degrees);
    }
    /**
     * @return {?}
     */
    get() {
        return this._croppie.get();
    }
}
NgxCroppieComponent.decorators = [
    { type: Component, args: [{
                // tslint:disable-next-line:component-selector
                selector: 'ngx-croppie',
                template: `<div #imageEdit (update)="newResult()"></div>`
            },] },
];
/** @nocollapse */
NgxCroppieComponent.propDecorators = {
    "imageEdit": [{ type: ViewChild, args: ['imageEdit',] },],
    "croppieOptions": [{ type: Input },],
    "points": [{ type: Input },],
    "outputFormatOptions": [{ type: Input },],
    "defaultZoom": [{ type: Input },],
    "result": [{ type: Output },],
    "imageUrl": [{ type: Input },],
};
function NgxCroppieComponent_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    NgxCroppieComponent.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    NgxCroppieComponent.ctorParameters;
    /** @type {!Object<string,!Array<{type: !Function, args: (undefined|!Array<?>)}>>} */
    NgxCroppieComponent.propDecorators;
    /** @type {?} */
    NgxCroppieComponent.prototype.imageEdit;
    /** @type {?} */
    NgxCroppieComponent.prototype.croppieOptions;
    /** @type {?} */
    NgxCroppieComponent.prototype.points;
    /** @type {?} */
    NgxCroppieComponent.prototype.outputFormatOptions;
    /** @type {?} */
    NgxCroppieComponent.prototype.defaultZoom;
    /** @type {?} */
    NgxCroppieComponent.prototype.result;
    /** @type {?} */
    NgxCroppieComponent.prototype._croppie;
    /** @type {?} */
    NgxCroppieComponent.prototype.imgUrl;
}
