// ngx-croppie/bundles/ngx-croppie.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('croppie'), require('@angular/common')) :
    typeof define === 'function' && define.amd ? define('ngx-croppie', ['exports', '@angular/core', 'croppie', '@angular/common'], factory) :
    (factory((global['ngx-croppie'] = {}),global.ng.core,null,global.ng.common));
}(this, (function (exports,core,Croppie,common) { 'use strict';

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var NgxCroppieComponent = /** @class */ (function () {
        function NgxCroppieComponent() {
            this.outputFormatOptions = { type: 'base64', size: 'viewport' };
            this.defaultZoom = 0;
            this.result = new core.EventEmitter();
        }
        Object.defineProperty(NgxCroppieComponent.prototype, "imageUrl", {
            get: /**
             * @return {?}
             */ function () {
                return this.imgUrl;
            },
            set: /**
             * @param {?} url
             * @return {?}
             */ function (url) {
                if (this.imgUrl === url) {
                    return;
                }
                this.imgUrl = url;
                if (this._croppie) {
                    this.bindToCroppie(this.imageUrl, this.points, this.defaultZoom);
                }
            },
            enumerable: true,
            configurable: true
        });
        /**
         * @return {?}
         */
        NgxCroppieComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
                this._croppie = new Croppie(this.imageEdit.nativeElement, this.croppieOptions);
                this.bindToCroppie(this.imageUrl, this.points, this.defaultZoom);
            };
        /**
         * @param {?} url
         * @param {?} points
         * @param {?} zoom
         * @return {?}
         */
        NgxCroppieComponent.prototype.bindToCroppie = /**
         * @param {?} url
         * @param {?} points
         * @param {?} zoom
         * @return {?}
         */
            function (url, points, zoom) {
                this._croppie.bind({ url: url, points: points, zoom: zoom });
            };
        /**
         * @return {?}
         */
        NgxCroppieComponent.prototype.newResult = /**
         * @return {?}
         */
            function () {
                var _this = this;
                this._croppie.result(this.outputFormatOptions).then(function (res) {
                    _this.result.emit(res);
                });
            };
        /**
         * @param {?} degrees
         * @return {?}
         */
        NgxCroppieComponent.prototype.rotate = /**
         * @param {?} degrees
         * @return {?}
         */
            function (degrees) {
                this._croppie.rotate(degrees);
            };
        /**
         * @return {?}
         */
        NgxCroppieComponent.prototype.get = /**
         * @return {?}
         */
            function () {
                return this._croppie.get();
            };
        NgxCroppieComponent.decorators = [
            { type: core.Component, args: [{
                        // tslint:disable-next-line:component-selector
                        selector: 'ngx-croppie',
                        template: "<div #imageEdit (update)=\"newResult()\"></div>"
                    },] },
        ];
        /** @nocollapse */
        NgxCroppieComponent.propDecorators = {
            "imageEdit": [{ type: core.ViewChild, args: ['imageEdit',] },],
            "croppieOptions": [{ type: core.Input },],
            "points": [{ type: core.Input },],
            "outputFormatOptions": [{ type: core.Input },],
            "defaultZoom": [{ type: core.Input },],
            "result": [{ type: core.Output },],
            "imageUrl": [{ type: core.Input },],
        };
        return NgxCroppieComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var NgxCroppieModule = /** @class */ (function () {
        function NgxCroppieModule() {
        }
        NgxCroppieModule.decorators = [
            { type: core.NgModule, args: [{
                        imports: [
                            common.CommonModule
                        ],
                        declarations: [NgxCroppieComponent],
                        exports: [
                            NgxCroppieComponent
                        ]
                    },] },
        ];
        return NgxCroppieModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */

    exports.NgxCroppieComponent = NgxCroppieComponent;
    exports.NgxCroppieModule = NgxCroppieModule;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
