// ngx-croppie/fesm2015/ngx-croppie.js
import { Component, Input, EventEmitter, Output, ViewChild, NgModule } from '@angular/core';
import * as Croppie from 'croppie';
import { CommonModule } from '@angular/common';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class NgxCroppieComponent {
    constructor() {
        this.outputFormatOptions = { type: 'base64', size: 'viewport' };
        this.defaultZoom = 0;
        this.result = new EventEmitter();
    }
    /**
     * @return {?}
     */
    get imageUrl() {
        return this.imgUrl;
    }
    /**
     * @param {?} url
     * @return {?}
     */
    set imageUrl(url) {
        if (this.imgUrl === url) {
            return;
        }
        this.imgUrl = url;
        if (this._croppie) {
            this.bindToCroppie(this.imageUrl, this.points, this.defaultZoom);
        }
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        this._croppie = new Croppie(this.imageEdit.nativeElement, this.croppieOptions);
        this.bindToCroppie(this.imageUrl, this.points, this.defaultZoom);
    }
    /**
     * @param {?} url
     * @param {?} points
     * @param {?} zoom
     * @return {?}
     */
    bindToCroppie(url, points, zoom) {
        this._croppie.bind({ url, points, zoom });
    }
    /**
     * @return {?}
     */
    newResult() {
        this._croppie.result(this.outputFormatOptions).then((res) => {
            this.result.emit(res);
        });
    }
    /**
     * @param {?} degrees
     * @return {?}
     */
    rotate(degrees) {
        this._croppie.rotate(degrees);
    }
    /**
     * @return {?}
     */
    get() {
        return this._croppie.get();
    }
}
NgxCroppieComponent.decorators = [
    { type: Component, args: [{
                // tslint:disable-next-line:component-selector
                selector: 'ngx-croppie',
                template: `<div #imageEdit (update)="newResult()"></div>`
            },] },
];
/** @nocollapse */
NgxCroppieComponent.propDecorators = {
    "imageEdit": [{ type: ViewChild, args: ['imageEdit',] },],
    "croppieOptions": [{ type: Input },],
    "points": [{ type: Input },],
    "outputFormatOptions": [{ type: Input },],
    "defaultZoom": [{ type: Input },],
    "result": [{ type: Output },],
    "imageUrl": [{ type: Input },],
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class NgxCroppieModule {
}
NgxCroppieModule.decorators = [
    { type: NgModule, args: [{
                imports: [
                    CommonModule
                ],
                declarations: [NgxCroppieComponent],
                exports: [
                    NgxCroppieComponent
                ]
            },] },
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

export { NgxCroppieComponent, NgxCroppieModule };
