// ngx-croppie/esm5/src/app/modules/ngx-croppie/ngx-croppie.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component, Input, EventEmitter, Output, ViewChild, ElementRef } from '@angular/core';
import * as Croppie from 'croppie';
var NgxCroppieComponent = /** @class */ (function () {
    function NgxCroppieComponent() {
        this.outputFormatOptions = { type: 'base64', size: 'viewport' };
        this.defaultZoom = 0;
        this.result = new EventEmitter();
    }
    Object.defineProperty(NgxCroppieComponent.prototype, "imageUrl", {
        get: /**
         * @return {?}
         */
        function () {
            return this.imgUrl;
        },
        set: /**
         * @param {?} url
         * @return {?}
         */
        function (url) {
            if (this.imgUrl === url) {
                return;
            }
            this.imgUrl = url;
            if (this._croppie) {
                this.bindToCroppie(this.imageUrl, this.points, this.defaultZoom);
            }
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    NgxCroppieComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this._croppie = new Croppie(this.imageEdit.nativeElement, this.croppieOptions);
        this.bindToCroppie(this.imageUrl, this.points, this.defaultZoom);
    };
    /**
     * @param {?} url
     * @param {?} points
     * @param {?} zoom
     * @return {?}
     */
    NgxCroppieComponent.prototype.bindToCroppie = /**
     * @param {?} url
     * @param {?} points
     * @param {?} zoom
     * @return {?}
     */
    function (url, points, zoom) {
        this._croppie.bind({ url: url, points: points, zoom: zoom });
    };
    /**
     * @return {?}
     */
    NgxCroppieComponent.prototype.newResult = /**
     * @return {?}
     */
    function () {
        var _this = this;
        this._croppie.result(this.outputFormatOptions).then(function (res) {
            _this.result.emit(res);
        });
    };
    /**
     * @param {?} degrees
     * @return {?}
     */
    NgxCroppieComponent.prototype.rotate = /**
     * @param {?} degrees
     * @return {?}
     */
    function (degrees) {
        this._croppie.rotate(degrees);
    };
    /**
     * @return {?}
     */
    NgxCroppieComponent.prototype.get = /**
     * @return {?}
     */
    function () {
        return this._croppie.get();
    };
    NgxCroppieComponent.decorators = [
        { type: Component, args: [{
                    // tslint:disable-next-line:component-selector
                    selector: 'ngx-croppie',
                    template: "<div #imageEdit (update)=\"newResult()\"></div>"
                },] },
    ];
    /** @nocollapse */
    NgxCroppieComponent.propDecorators = {
        "imageEdit": [{ type: ViewChild, args: ['imageEdit',] },],
        "croppieOptions": [{ type: Input },],
        "points": [{ type: Input },],
        "outputFormatOptions": [{ type: Input },],
        "defaultZoom": [{ type: Input },],
        "result": [{ type: Output },],
        "imageUrl": [{ type: Input },],
    };
    return NgxCroppieComponent;
}());
export { NgxCroppieComponent };
function NgxCroppieComponent_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    NgxCroppieComponent.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    NgxCroppieComponent.ctorParameters;
    /** @type {!Object<string,!Array<{type: !Function, args: (undefined|!Array<?>)}>>} */
    NgxCroppieComponent.propDecorators;
    /** @type {?} */
    NgxCroppieComponent.prototype.imageEdit;
    /** @type {?} */
    NgxCroppieComponent.prototype.croppieOptions;
    /** @type {?} */
    NgxCroppieComponent.prototype.points;
    /** @type {?} */
    NgxCroppieComponent.prototype.outputFormatOptions;
    /** @type {?} */
    NgxCroppieComponent.prototype.defaultZoom;
    /** @type {?} */
    NgxCroppieComponent.prototype.result;
    /** @type {?} */
    NgxCroppieComponent.prototype._croppie;
    /** @type {?} */
    NgxCroppieComponent.prototype.imgUrl;
}
