// ngx-croppie/fesm5/ngx-croppie.js
import { Component, Input, EventEmitter, Output, ViewChild, NgModule } from '@angular/core';
import * as Croppie from 'croppie';
import { CommonModule } from '@angular/common';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var NgxCroppieComponent = /** @class */ (function () {
    function NgxCroppieComponent() {
        this.outputFormatOptions = { type: 'base64', size: 'viewport' };
        this.defaultZoom = 0;
        this.result = new EventEmitter();
    }
    Object.defineProperty(NgxCroppieComponent.prototype, "imageUrl", {
        get: /**
         * @return {?}
         */
        function () {
            return this.imgUrl;
        },
        set: /**
         * @param {?} url
         * @return {?}
         */
        function (url) {
            if (this.imgUrl === url) {
                return;
            }
            this.imgUrl = url;
            if (this._croppie) {
                this.bindToCroppie(this.imageUrl, this.points, this.defaultZoom);
            }
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    NgxCroppieComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this._croppie = new Croppie(this.imageEdit.nativeElement, this.croppieOptions);
        this.bindToCroppie(this.imageUrl, this.points, this.defaultZoom);
    };
    /**
     * @param {?} url
     * @param {?} points
     * @param {?} zoom
     * @return {?}
     */
    NgxCroppieComponent.prototype.bindToCroppie = /**
     * @param {?} url
     * @param {?} points
     * @param {?} zoom
     * @return {?}
     */
    function (url, points, zoom) {
        this._croppie.bind({ url: url, points: points, zoom: zoom });
    };
    /**
     * @return {?}
     */
    NgxCroppieComponent.prototype.newResult = /**
     * @return {?}
     */
    function () {
        var _this = this;
        this._croppie.result(this.outputFormatOptions).then(function (res) {
            _this.result.emit(res);
        });
    };
    /**
     * @param {?} degrees
     * @return {?}
     */
    NgxCroppieComponent.prototype.rotate = /**
     * @param {?} degrees
     * @return {?}
     */
    function (degrees) {
        this._croppie.rotate(degrees);
    };
    /**
     * @return {?}
     */
    NgxCroppieComponent.prototype.get = /**
     * @return {?}
     */
    function () {
        return this._croppie.get();
    };
    NgxCroppieComponent.decorators = [
        { type: Component, args: [{
                    // tslint:disable-next-line:component-selector
                    selector: 'ngx-croppie',
                    template: "<div #imageEdit (update)=\"newResult()\"></div>"
                },] },
    ];
    /** @nocollapse */
    NgxCroppieComponent.propDecorators = {
        "imageEdit": [{ type: ViewChild, args: ['imageEdit',] },],
        "croppieOptions": [{ type: Input },],
        "points": [{ type: Input },],
        "outputFormatOptions": [{ type: Input },],
        "defaultZoom": [{ type: Input },],
        "result": [{ type: Output },],
        "imageUrl": [{ type: Input },],
    };
    return NgxCroppieComponent;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var NgxCroppieModule = /** @class */ (function () {
    function NgxCroppieModule() {
    }
    NgxCroppieModule.decorators = [
        { type: NgModule, args: [{
                    imports: [
                        CommonModule
                    ],
                    declarations: [NgxCroppieComponent],
                    exports: [
                        NgxCroppieComponent
                    ]
                },] },
    ];
    return NgxCroppieModule;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

export { NgxCroppieComponent, NgxCroppieModule };
